examples/seismic/acoustic/operators.py

```diff
@@ -144,10 +146,28 @@ def ForwardOperator(model, geometry, space_order=4,
 
     # Create interpolation expression for receivers
     rec_term = rec.interpolate(expr=u)
-
+    # Build operator equations
+    equations = eqn + src_term + rec_term
+
+    if factor:
+        # Implement snapshotting
+        nsnaps = (geometry.nt + factor - 1) // factor
+        time_subsampled = ConditionalDimension(
+            't_sub', parent=model.grid.time_dim, factor=factor)
+        usnaps = TimeFunction(name='usnaps', grid=model.grid,
+                              time_order=2, space_order=space_order,
+                              save=nsnaps, time_dim=time_subsampled)
+        # Add equation to save snapshots
+        snapshot_eq = Eq(usnaps, u)
+        equations += [snapshot_eq]
+    else:
+        usnaps = None
     # Substitute spacing terms to reduce flops
-    return Operator(eqn + src_term + rec_term, subs=model.spacing_map,
-                    name='Forward', **kwargs)
+    op = Operator(equations, subs=model.spacing_map, name='Forward', **kwargs)
+    if usnaps is not None:
+        return op, usnaps
```

Comment:
No the operator build cannot return objects like that. This is an abstract operator with placeholders that might not be correct for runtime.

Reply:
Fixed. The operator build only returns op now.